### PR TITLE
Check ssh key is in ssh-agent

### PR DESCRIPTION
**Pull request by natalia-miheyeva**
Added an extra check in ssh installed? method. This fix would only be useful for somebody who generated SSH key and not added it to the ssh-agent prior to running clio_setup.

By just doing `ssh-add -l` to list the keys was not sufficient since ssh-agent keeps the keys in memory and after rebooting running `ssh-add -l` would return 'The agent has no identities.'. So I made a decision to run `ssh-add -K` as there is no harm in doing so.

### CODE REVIEW

File: lib/clio_setup/component/ssh.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class Component::Ssh < Component
     KEY = "#{ENV["HOME"]}/.ssh/id_rsa.pub"
 
     def installed?
-      File.exist?(KEY)
+      File.exist?(KEY) && (system("ssh-add -K >/dev/null 2>&1; ssh-add -l >/dev/null"))
```
Comment (mohsen3) [inline lib/clio_setup/component/ssh.rb]: `(system("ssh-add -K >/dev/null 2>&1; ssh-add -l >/dev/null"))`
What does this mean? I'd add a comment.

File: lib/clio_setup/component/ssh.rb [hunk 2]
```diff
@@ -16,11 +18,14 @@ def name
 
     def install
       FileUtils.mkdir_p("#{ENV["HOME"]}/Library/LaunchAgents")
-
       puts "\nYou will now be prompted for a password for your SSH key. Don't make one up!"
       puts "Use LastPass to generate a password, then paste it in here. (You'll be prompted"
       puts "for the password three times.)\n\n"
+      puts "\nStart the ssh-agent in the background.\n"
+      run("eval \"$(ssh-agent -s)\"")
```
Comment (fimmtiu) [inline lib/clio_setup/component/ssh.rb]: This won't do quite what you expect, unfortunately; it'll spin up a new subshell to run this `eval` command, which sets the environment variables in that subshell, and then that subshell exits. The environment variable changes won't get propagated back up to the `clio_setup` script.

This is good stuff, though! I've taken your changes and implemented them differently here: https://github.com/fimmtiu/clio_setup/pull/6 . Please have a look and see what you think.

Thanks for your help!

File: lib/clio_setup/component/ssh.rb [hunk 3]
```diff
@@ -2,8 +2,10 @@ module ClioSetup
   class Component::Ssh < Component
     KEY = "#{ENV["HOME"]}/.ssh/id_rsa.pub"
 
+    # Check if the id_rsa.pub file exists and check that the key is successfully
+    # added to the keychain
     def installed?
-      File.exist?(KEY)
+      File.exist?(KEY) && (system("ssh-add -K >/dev/null 2>&1; ssh-add -l >/dev/null"))
```
Comment (fimmtiu) [inline lib/clio_setup/component/ssh.rb]: In general, I'm of the opinion that predicate methods like `installed?` should just return true or false without causing any side effects. It's a good idea to check that the key is installed here, though. I've implemented this check in a slightly different way in https://github.com/fimmtiu/clio_setup/pull/6.

**Comment by natalia-miheyeva** [comment]
This was resolved in https://github.com/fimmtiu/clio_setup/pull/6. Closing

**Pull request closed without merging**